### DISCUSSION TITLE
support creation of `package.json`

### DISCUSSION THREAD

**Discussion by HKhademian** [oven-sh/bun#698] [Ideas]
Currently when I run `bun i struktjs` in an empty folder, it creates an error :
`error: MissingPackageJSON
bun could not find a package.json file.`
Which means, it requires to create `package.json` file, before any package installs.

To generate that file, we must use pnpm, npm, or yarn.
Another way is to use `bun create blank .` which generates a bootstrap project structure.
In my taste is not a good match for the problem.

But why not bun provides a simple way, like `bun init`, similar to `bun create` and pipe this command (like `bun create`) internally to those PMs.

ALSO Bun clams that [Bun's a package manager](https://github.com/oven-sh/bun#using-bun-as-a-package-manager), so like others it has to provide a way to init basic PM requirements.

I suggest to add `bun init [optional template from create] <path>` command.
If both args provided, just use `bun create` subcommand.
When no template provided (maybe an arg provided as path), it just inits `package.json` (maybe `tsconfig.json`)
and when no path provided (no args), it uses current directory for above.

```console
bun init next   my-best-next-project
bun init blank  bun-blank-project
bun init        my-unicorn
bun init        /special-path/to/project
bun init
```

**Comment by AroLeaf** [comment]
> Currently when I run `bun i struktjs`

The command to add a dependency is `bun add`/`bun a`, which should work without a `package.json` present.
